src/deployers/BasicPluginsDeployer.cpp

```diff
@@ -29,7 +29,21 @@ BasicPluginsDeployer::BasicPluginsDeployer(std::string moduleName,
                                                                   qtDataPath(std::move(qtDataPath)) {}
 
 bool BasicPluginsDeployer::deploy() {
-    // currently this is a no-op, but we might add more functionality later on, such as some kinds of default
-    // attempts to copy data based on the moduleName
+    for (const auto &pluginName : qtPluginsToBeDeployed()) {
```

Comment:
It's a nice cleanup, but instead of introducing this code here along with the `qtPluginsToBeDeployed` function, I would suggest to introduce a protected `BasicPluginsDeployer::deployPlugins(const std::vector<std::string> &plugins)`, so in each `deploy()` implementation one could just do for example:
```c++
    deployPlugins({"bearer"});
```
I think that's adding a bit less complexity overall, since it removes the need for both `qtPluginsToBeDeployed` and `customDeploy`.

Reply:
while a `deployStandardQtPlugins()` could be nice indeed having a `customDeploy()` or if going with this function then maybe `doDeploy()` is a better naming, this is because we override methods and current code was calling the base method which makes the code slower due extra branching and is error prone as one has to remember to call that.
So I'll add a new commit and we see what is best.

Reply:
Yeah, that's better and I do think it's nice how `doDeploy` allows to get rid of all the calls to the superclass. Though I would personally just have removed them, along with the base implementation, because "we might add more functionality later on" is not a good reason for all that cruft in my opinion.

Anyway, this patch is a good clean-up as-is in my opinion.